File: apps/node/src/fizzbuzz/application/fizz-buzz-list-command.ts
import { FizzBuzzList } from "../domain/model/fizz-buzz-list.js";
import { FizzBuzzType } from "../domain/type/fizz-buzz-type.js";
import { FizzBuzzCommand } from "./fizz-buzz-command.js";

export class FizzBuzzListCommand implements FizzBuzzCommand {
  private readonly _type: FizzBuzzType;
  private readonly _count: number;

  constructor(type: FizzBuzzType, count: number) {
    this._type = type;
    this._count = count;
  }

  execute(): FizzBuzzList {
    return Array.from({ length: this._count }, (_, i) => i + 1).reduce(
      (list, num) => list.add(this._type.generate(num)),
      new FizzBuzzList(),
    );
  }
}

File: apps/csharp/FizzBuzz/Application/FizzBuzzListCommand.cs
namespace FizzBuzz.Application;

using FizzBuzz.Domain.Model;
using FizzBuzz.Domain.Type;

public sealed class FizzBuzzListCommand : IFizzBuzzCommand
{
    private readonly IFizzBuzzType _type;

    public FizzBuzzListCommand(IFizzBuzzType type)
    {
        _type = type;
    }

    public FizzBuzzValue ExecuteValue(int number)
    {
        throw new NotSupportedException(
            "FizzBuzzListCommand does not support single value execution.");
    }

    public FizzBuzzList ExecuteList(int count)
    {
        var values = Enumerable.Range(1, count)
            .Select(i => _type.Generate(i))
            .ToList();

        return new FizzBuzzList(values);
    }
}
